scripts/host-check

```diff
@@ -1169,6 +1181,87 @@ def check_shmem_pages_max_size() -> CheckFuncReturn:
     return CheckState.SUCCESS, f"{shared_mem_max_page_size:.1f} GiB"
 
 
+def _check_module_param(
+    module: str, exp_param_vals: Dict[str, str]
+) -> Dict[str, Optional[str]]:
+    """
+    For a kernel module, check if the runtime parameters have the expected
+    values.
+
+    :param module:
+        Name of the module
+
+    :param exp_param_vals:
+        Dictionary mapping parameter name to the expected value of the parameter
+
+    :return:
+        A dict containing parameters with unexpected values
+        If failed to check value for a parameter, returns None as the dict
+        value for the parameter
+    """
+    param_unexpected: Dict[str, Optional[str]] = {}
+
+    if not _is_module_loaded(
+        module.replace("-", "_")
+    ) and not _is_module_builtin(module):
+        # Skip module param check if module is not loaded
+        return param_unexpected
+    # sysfs filesystem requires the underscore version of the kernel module
+    # name
+    module = module.replace("-", "_")
+    all_params_str, _ = run_cmd(["modinfo", "--field", "parm", module])
+    for param, exp_val in exp_param_vals.items():
+        if not re.search(rf"^{param}:", all_params_str, re.MULTILINE):
+            # if 'param' is absent in modinfo output, it means either the
+            # parameter was introduced in a later kernel version, or the
+            # specific linux distribution does not support the parameter.
+            continue
+        try:
+            path = f"/sys/module/{module}/parameters/{param}"
+            with open(path, "r", encoding="utf-8") as f:
+                param_val = f.read().strip()
+            # (null) implies the parameter isn't set, and is thus the
+            # default value
+            if param_val not in (exp_val, "(null)"):
+                param_unexpected[param] = param_val
+        except FileNotFoundError:
+            param_unexpected[param] = None
+
+    return param_unexpected
+
+
+def check_module_non_default_params() -> CheckFuncReturn:
```

Comment:
I wouldn't have `non_default` in the function name here - just calling it `check_module_params` seems better.

Reply:
Had `non_default` in the name because this function mainly just creates the string message, The only function `_check_module_param` checks the parameters against expected params. Since this function creates messages like:
"XRd has only been tested with default kernel module parameters." or "Kernel modules loaded with default parameters". I think it should have "default" in its name

Reply:
Or the other option is changing those messages. Should I change them to eg 
 "The default value for parameter ..."-> "The expected value for parameter ..."
"XRd has only been tested with default kernel module parameters." -> "XRd has not been tested with set kernel module parameters."

Reply:
Yes I think change 'default' to 'expected' in these places.

 I still think the name of the function should be positive - i.e. 'check_module_expected_params'